Task 0 done

IvanPavlenk0 opened the pull request.

IvanPavlenk0 committed:
Task 0 done

IvanBesarab commented on Assignment2/Assignment2.xcodeproj/xcuserdata/AlbroVan.xcuserdatad/xcschemes/xcschememanagement.plist:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>SchemeUserState</key>
+	<dict>
+		<key>Assignment2.xcscheme_^#shared#^_</key>
+		<dict>
+			<key>orderHint</key>
+			<integer>0</integer>
+		</dict>
```
Do not include in future PR's

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,7 +12,12 @@ class ViewController: RootViewController {
     //    Задача 0. Вывести наибольшее из двух чисел
     func findGreatestNumber(numberA: Int, numberB: Int) {
-        print("Greatest number is 9")
+        if numberA > numberB {
+            print("Greatest number is \(numberA)")
+        }
+        else {
+            print("Greatest number is \(numberB)")
+        }
```
Make code 5 times shorter

IvanPavlenk0 committed:
Update task 0

IvanPavlenk0 committed:
Task 1 - 3

Task 0 - code is shorter
Tack 1 - done
Task 2 and 3 - don't understand why code doesn't work

IvanPavlenk0 committed:
All done

Task 0 -> code is shorter
Task 1 - 5 -> first show

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     //    Задача 1. Вывести квадрат и куб введенного числа
     func findSquareAndQubeOf(number: Int) {
-        fatalError()
+        let a = number * number
```
Watch your naming, use result as default variable name

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     //    Задача 0. Вывести наибольшее из двух чисел
     func findGreatestNumber(numberA: Int, numberB: Int) {
-        print("Greatest number is 9")
+        print("Greatest number is \(numberA > numberB ? numberA : numberB)")
```
:fire:

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
-        fatalError()
+        let a = number * number
+        print("""
+square of \(number) = \(a)
+qube of \(number) = \(a * number)
```
Use two separate functions square() and cube()

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     func findSquareAndQubeOf(number: Int) {
-        fatalError()
+        let a = number * number
+        print("""
```
Use single print for each function return

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     func numbersUpToNAndBackToZero(with number: Int) {
-        fatalError()
+        var counter = 1
+        for i in 0...number + number {
+            if i > number {
+                counter += 1
+                print(i - counter)
+                counter += 1
+            }
+            else {
+                print(i)
+            }
```
Cool

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     //    Задача 3. Подсчитать общее число делителей числа и вывести их
     func findDivisors(of number: Int) {
-        fatalError()
+        var divisorsCount = 0
+        for i in 1..<number {
+            if number % i == 0 {
+                print(i)
+                divisorsCount += 1
+            }
+        }
+        print("number \(number) has \(divisorsCount) divisors")
```
Cool

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
     //    Задача 4 Проверить, является ли заданное число совершенным
     func isNumberPerfect(_ number: Int) {
-        fatalError()
+        var divisorsCount = 0
+        for i in 1..<number {
+            if number % i == 0 {
+                divisorsCount += i
+            }
+        }
+        if number == divisorsCount {
+            print("number \(number) is perfect")
+        }
+        else {
+            print("number \(number) is NOT perfect")
+        }
```
Make a func that helps you not duplicate code in this and prev task

IvanBesarab commented on Assignment2/Assignment2/ViewController.swift:
```diff
@@ -12,33 +12,85 @@ class ViewController: RootViewController {
 	func mirror(number: Int) {
-		fatalError()
-	}
+        var num = number
+        var lengthOfNumber = 1
+        var multiplier = 10
+        while multiplier < num {
+            lengthOfNumber += 1
+            multiplier *= 10
+        }
+        var reNumber = 0
+        var reNumberCounter = 0
+        while multiplier > num {
+            if num % 10 != 0 {
+                num -= 1
+                reNumberCounter += 1
+            } else {
+                num /= 10
+                multiplier /= 10
+                reNumber = reNumberCounter * multiplier + reNumber
+                reNumberCounter = 0
+            }
+        }
+        print ("Mirrored number is \(reNumber)")
```
Can be done with single while loop == twice shorter

IvanPavlenk0 committed:
Optimized assignment 2
